crates/katana/rpc/rpc-types/src/transaction.rs

```diff
@@ -99,13 +128,39 @@ impl BroadcastedDeclareTx {
                     }),
                 })
             }
+
+            BroadcastedDeclareTransaction::V3(tx) => {
+                // TODO: avoid computing the class hash again
```

Comment:
What's the alternatives to do so? Adding some fields instead of the new type pattern to have it store once and never recomputed during the life of the instanciated object?

Reply:
Subsequent PR then?

Reply:
forgot to reply to this lol. Yeah it's just a copy pasted impl but we should definitely improve how we convert the contract classes. Let's do a follow up PR for this.